src/lightning_app/components/auto_scaler.py

```diff
@@ -0,0 +1,529 @@
+import asyncio
+import logging
+import os
+import secrets
+import time
+import uuid
+from base64 import b64encode
+from itertools import cycle
+from typing import Any, Dict, List, Optional, Tuple
+
+import aiohttp
+import aiohttp.client_exceptions
+import requests
+import uvicorn
+from fastapi import Depends, FastAPI, HTTPException, Request
+from fastapi.middleware.cors import CORSMiddleware
+from fastapi.responses import RedirectResponse
+from fastapi.security import HTTPBasic, HTTPBasicCredentials
+from lightning_utilities.core.overrides import is_overridden
+from pydantic import BaseModel
+from starlette.status import HTTP_401_UNAUTHORIZED
+
+from lightning_app.core.flow import LightningFlow
+from lightning_app.core.work import LightningWork
+from lightning_app.utilities.app_helpers import Logger
+from lightning_app.utilities.packaging.cloud_compute import CloudCompute
+
+logger = Logger(__name__)
+
+
+def _raise_granular_exception(exception: Exception) -> None:
+    """Handle an exception from hitting the model servers."""
+    if not isinstance(exception, Exception):
+        return
+
+    if isinstance(exception, HTTPException):
+        raise exception
+
+    if isinstance(exception, aiohttp.client_exceptions.ServerDisconnectedError):
+        raise HTTPException(500, "Worker Server Disconnected") from exception
+
+    if isinstance(exception, aiohttp.client_exceptions.ClientError):
+        logging.exception(exception)
+        raise HTTPException(500, "Worker Server error") from exception
+
+    if isinstance(exception, asyncio.TimeoutError):
+        raise HTTPException(408, "Request timed out") from exception
+
+    if isinstance(exception, Exception):
+        if exception.args[0] == "Server disconnected":
+            raise HTTPException(500, "Worker Server disconnected") from exception
+
+    logging.exception(exception)
+    raise HTTPException(500, exception.args[0]) from exception
+
+
+class _SysInfo(BaseModel):
+    num_workers: int
+    servers: List[str]
+    num_requests: int
+    process_time: int
```

Comment:
is it start time or how long does one prediction take?
```suggestion
    processing_time: int
```

Reply:
It is a period of time but not a moment in time. SGTM!